### PR TITLE
correctly filter past/upcoming rides for rider

### PR DESCRIPTION
**Pull request by harrisonchin2523**
### Summary 
There was an issue with the rider aspect of web where past rides would not filter correctly and would still be displayed in the "Your Upcoming Rides" tab. These changes should fix this.

### Notes 
There seems to be an issue with the isPast condition when filtering rides between Schedule.tsx and RiderScheduleTable.tsx. It is not correctly filtering rides that have come to pass. I'm not exactly sure why this is happening, but I wanted to implement this fix so we could correctly filter rides and come back to this after implementing testing.

### CODE REVIEW

**Review by pratyush1712 — COMMENTED** [review]
Thank you for noticing this issue and fixing it. Can you also share some screenshots or screen recordings explaining the problem and how this pr will fix it? Thanks

**Comment by harrisonchin2523** [comment]
@pratyush1712 here are two screen recordings of the before and after:
- Before:
https://user-images.githubusercontent.com/88568615/222927753-695562f1-3339-48b4-b927-8e54eadac22c.mov
- After:
https://user-images.githubusercontent.com/88568615/222927768-f869ee0d-64ec-4a88-9bc7-ee0252eb0606.mov

File: frontend/src/pages/Rider/Schedule.tsx [hunk 1]
```diff
@@ -17,7 +17,10 @@ import styles from './page.module.css';
 
 const Schedule = () => {
   const componentMounted = useRef(true);
-  const [rides, setRides] = useState<Ride[]>();
+  const now = new Date().toISOString();
+  const [rides, setRides] = useState<Ride[]>([]);
```
Comment (pratyush1712) [inline frontend/src/pages/Rider/Schedule.tsx:21]: Good work! These changes work great!

**Review by pratyush1712 — APPROVED** [review]
Good job! Thanks for noticing the issue and fixing it!

**Pull request merged**